llvm/unittests/CAS/CASUtilsTest.cpp

```diff
@@ -0,0 +1,56 @@
+//===- CASUtilsTest.cpp ---------------------------------------------------===//
+//
+// Part of the LLVM Project, under the Apache License v2.0 with LLVM Exceptions.
+// See https://llvm.org/LICENSE.txt for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+//
+//===----------------------------------------------------------------------===//
+
+#include "llvm/CAS/CASDB.h"
+#include "llvm/CAS/Utils.h"
+#include "gtest/gtest.h"
+
+using namespace llvm;
+using namespace llvm::cas;
+
+TEST(CASUtilsTest, walkFileTreeRecursively) {
+  std::unique_ptr<CASDB> CAS = createInMemoryCAS();
+
+  auto make = [&](StringRef Content) {
+    return *expectedToOptional(CAS->createBlob(Content));
+  };
+
+  HierarchicalTreeBuilder Builder;
+  Builder.push(make("blob2"), TreeEntry::Regular, "/d2");
```

Comment:
I know you're copying my pattern here but I don't love that `*expectedToOptional()` gets used (I'd like to clean up the tests I wrote at some point...).

Since this is an in-memory CAS, which really isn't going to fail, it's probably best to just use `cantFail(CAS->createBlob(Content))`.

But the following maybe isn't too bad and would print the error in the unlikely event of failure:
```
HierarchicalTreeBuilder Builder;
auto push = [&](StringRef Content, TreeEntry::Kind K, StringRef Name) {
  Expected<BlobRef> Blob = CAS->createBlob(Content);
  if (Blob)
   Builder.push(*Blob, K, Name);
  return Blob.takeError();
};
ASSERT_THAT_ERROR(push("blob2", TreeEntry::Regular, "/d2"), Succeeded());
```
(I suggest the former option, as mentioned, but showing the other option in case it's useful elsewhere.)